src/shared_modules/utils/evpHelper.hpp

```diff
@@ -0,0 +1,140 @@
+/*
+ * Wazuh shared modules utils
+ * Copyright (C) 2015, Wazuh Inc.
+ * July 10, 2024.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef _EVPHELPER_HPP
+#define _EVPHELPER_HPP
+
+#include "defer.hpp"
+#include <array>
+#include <opensslPrimitives.hpp>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+constexpr auto CIPHER_KEY_SIZE {32};
+constexpr auto CIPHER_IV_SIZE {16};
+constexpr auto OPENSSL_SUCCESS {1};
+
+template<typename T = OpenSSLPrimitives>
+class EVPHelper final : public T
+{
+public:
+    explicit EVPHelper() = default;
+    ~EVPHelper() override = default;
+
+    /**
+     * Encrypts the input vector with the provided key
```

Comment:
We are not providing any key
```suggestion
     * Encrypts the input vector
```